Adds the option to jump directly to the pageview after the search.

frank-ulrich-weber opened the pull request:
In case of exactly one search result the search plugin can be configured
to open the pageview instead of the listview.

frank-ulrich-weber committed:
Adds the option to jump directly to the pageview after the search.

In case of exactly one search result the search plugin can be configured
to open the pageview instead of the listview.

sebastian-meyer commented on dlf/plugins/search/class.tx_dlf_search.php:
```diff
@@ -654,16 +654,31 @@ public function main($content, $conf) {
-						), '', TRUE, FALSE);
+					array (
+						'id' => $results->current()['uid'],
+						'page' => 1
```
If the only search result was a fulltext hit, the target should be a specific page, not just page 1.

I suggest checking if the search result references a page number and then using this - otherwise page 1.

sebastian-meyer reviewed (changes requested):
Apart from one little suggestion this looks good to me!

frank-ulrich-weber committed:
displyCond tergetPidPageView corrected

frank-ulrich-weber committed:
In the case of one search result:

Only if there is one subpart, set 'page' to the page of the hit,
otherwise to 1.

Independently set the URL parameter 'highlight_word'.

sebastian-meyer commented on dlf/plugins/search/flexform.xml:
```diff
@@ -220,6 +220,17 @@
+							<label>LLL:EXT:dlf/plugins/search/locallang.xml:tt_content.pi_flexform.showSingleResult</label>
+							<config>
+								<type>check</type>
+								<default>1</default>
```
I'd suggest turning this off as default. That is to prevent someone updating the extension from accidently activating a new feature he/she doesn't want. Also, the way the TYPO3 flexform configuration works, the configuration would have to be rewritten to actually change the settings although it would be falsely displayed as activated.

frank-ulrich-weber replied:
I see... then I change the logic "Show single result in listview" to "Show single result in pageview". Then the new feature can be disabled by defult.

sebastian-meyer commented on dlf/plugins/search/flexform.xml:
```diff
@@ -240,6 +251,21 @@
 					</targetPid>
+					<targetPidPageView>
+						<TCEforms>
+							<displayCond>FIELD:showSingleResult:=:0</displayCond>
```
This option should be shown when `showSingleResult` is set to `true`, right? Then its `displayCond` should be `FIELD:showSingleResult:=:1`.

frank-ulrich-weber replied:
No, it's the other way round. The change to "Show single result in pageview" fixes this, too.

sebastian-meyer reviewed (changes requested):
There is another little thing to consider.

frank-ulrich-weber committed:
Reversed List View by Page View and turned "jump to Page View" feature
of by default
